master/test/streamdata/stream_data.go

```diff
@@ -37,9 +49,192 @@ func GenerateStreamTrials() StreamTrialsData {
 	}
 	return StreamTrialsData{
 		MustMigrate: mustMigrate,
-		ExpID:       0,
-		JobID:       "test_job",
+		ExpID:       1,
+		JobID:       "test_job1",
 		TaskIDs:     []string{"1.1", "1.2", "1.3"},
 		TrialIDs:    []int{1, 2, 3},
 	}
 }
+
+type trial struct {
+	bun.BaseModel `bun:"table:trials"`
+	ID            int            `bun:"id,pk"`
+	ExperimentID  int            `bun:"experiment_id"`
+	HParams       map[string]any `bun:"hparams"`
+	State         model.State    `bun:"state"`
+	StartTime     time.Time      `bun:"start_time"`
+	Seq           int64          `bun:"seq"`
+}
+
+func ModTrial(ctx context.Context, trialID, experimentID int, changeStart bool, changeSeq bool, changeState string) error {
+	trials, err := queryTrials(ctx)
+	if err != nil {
+		return err
+	}
+	startTime := time.Time{}
+	seq := int64(0)
+	maxSeq := int64(0)
+	state := model.State("")
+	for _, t := range trials {
+		if t.ID == trialID {
+			startTime = t.StartTime
+			seq = t.Seq
+			state = t.State
+		}
+		if t.Seq > maxSeq {
+			maxSeq = t.Seq
+		}
+	}
+
+	if changeStart {
+		startTime = time.Now()
+	}
+	if changeSeq {
+		seq = maxSeq + 1
+	}
+	if changeState != "" {
+		state = model.State(changeState)
+	}
+
+	_, err = db.Bun().NewUpdate().Model(&trial{
+		ID:           trialID,
+		ExperimentID: experimentID,
+		HParams:      nil,
+		State:        state,
+		StartTime:    startTime,
+		Seq:          seq,
+	}).Where("id = ?", trialID).Exec(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func queryTrials(ctx context.Context) ([]trial, error) {
+	var trials []trial
+	err := db.Bun().NewSelect().
+		Model(&trials).
+		Scan(ctx, &trials)
+	if err != nil {
+		return nil, err
+	}
+	return trials, nil
+}
+
+func AddTrial(ctx context.Context, experimentID int) error {
```

Comment:
question (non-blocking): would it be worth while to adjust these trial-specific functions so they could be re-used for other types? 
Like:
```
func queryStreamableType(ctx context.Context, entities *[]StreamableEntity) ( error error) {
    err = db.Bun().NewSelect().Model(entities).Scan(ctx, entities)
    ...
}
```

Reply:
hmmm I feel like in this case, since trials and experiments require multiple tables to be changed, I think that would be difficult to turn generic

Reply:
That's a fair point, my concern is that we might create a lot of these niche functions that all do relatively similar things and not get much re-use out of them.